### ISSUE TITLE
Scala.Js / Node.Js support

### ISSUE THREAD

**Issue by mkotsbak** [slick/slick#1313]

**Comment by elacin** [comment]
Just to chime in - I already wanted this. My use case was that i wanted to manipulate a slick `Query` in the frontend for a generic slick GUI i'm writing.
Also, being able to run a `Query` like that against something databasey in the browser, while sending that same query to backend(s) for execution against a proper database could be pretty neat!

**Comment by cvogt** [comment]
I think you are both talking about different use cases. First is executing queries directly on a js vm. Second is sending queries from a js client to a server vm and running them there. Both sound fun. I thought about the second one before and remember @lihaoyi had it too on Twitter. In any case, I don't have any js project myself at this point and I doubt this is high enough priority for Typesafe to put it's limited Slick resources on rather than on something else. But Slick is open source and this sounds like a good community project. So give it a shot, if you find the time :)

**Comment by mkotsbak** [comment]
Yes, it could be tried. Will probably stop on dependency on JDBC stuff that needs to be separated.

Regarding @elacin's idea, it sounds a bit scary to trust queries coming from the client, but maybe in some cases it could be safe, or just use it for parts of the query.

**Comment by elacin** [comment]
well its certainly applicable in all the situations where it would be acceptable to accept pure sql strings from a client - hidden well inside your corporate network. I would also expect this to meet jdbc pretty quickly, which is why i havent tried.

**Comment by triggerNZ** [comment]
Loosening up the JDBC dependency would help a lot on the JVM too, supporting databases like Hive, whose API is not JDBC

**Comment by mkotsbak** [comment]
Yes, JDBC is really something that you want to avoid if there are alternatives, as it is a blocking API...

**Comment by mkotsbak** [comment]
I would assume that JDBC is almost not used at all before the queries are actually executed, so you should be able to do all up to the DBIO actions are created in the JS part. It would require that part to be in a separate module.

Then another use case: accessing sqlite using React Native with Slick. See this project that allows the use of sqlite from React Native: https://github.com/andpor/react-native-sqlite-storage That library could actually be used instead of JDBC to run the actual SQL Slick would generate.
